.github/workflows/posix.yml

```diff
@@ -138,6 +143,12 @@ jobs:
       # xsel for clipboard tests
       run: sudo apt-get update && sudo apt-get install -y libc6-dev-i386 xsel ${{ env.EXTRA_APT }}
 
+    - name: Generate extra locales
+      # These extra locales will be available for locale.setlocale() calls in tests
+      run: |
+        sudo locale-gen ${{ env.EXTRA_LOC }}
```

Comment:
Does this mean the `Locale: it_IT.utf8` isnt working either?

Reply:
It depends what you mean by "working". Do you refer to the bug #46319 or to something more general ?

 - The issue #46319 happens only when the current locale uses a non-unicode encoding (no utf-8). So it does not happen in any of the workers by default
 - So in the test fixture parameters I added the `zh_CN` locale, that has a non-utf-8 encoding. 
 - Unfortunately this particular parameter value was never used in any of the workers (the test was skipped), since `locale.setlocale` was raising an error (locale not available) 
 - I finally found a way to make the locale available on linux workers: `sudo locale-gen <locale>`

Does that answer your question ?

Reply:
I think it was assumed setting the environment variables 

```
      LANG: ${{ matrix.lang || '' }}
      LC_ALL: ${{ matrix.lc_all || '' }}
```

would ensure the locale for the test run would not be the default (en_US.UTF-8) for the other locale builds.

So IIUC, this is necessary if setting a non-utf8 locale?

Reply:
Yes, unfortunately when I set `LANG` and `LC_ALL` to `zh_CN` (non-utf8), then there is a strange segmentation fault appearing  (see https://github.com/pandas-dev/pandas/pull/46405#issuecomment-1072773630)

So this is why I do not set it through these env variables but I only install it.

Reply:
If we're setting the locale this way, can we remove setting the `LANG` and `LC_ALL` environment variables here?